cli/integration/tests/waiter/test_cli.py

```diff
@@ -2352,3 +2352,85 @@ def test_stop_ping_service_force_kill(self):
+                      min_instances=1, multiple_services=False,
+                      test_service=False, test_instance=False):
+        token_name = self.token_name()
+        token_fields = util.minimal_service_description()
+        token_fields['min-instances'] = min_instances
+        try:
+            if multiple_services:
+                token_new_fields = util.minimal_service_description()
+                util.post_token(self.waiter_url, token_name, token_new_fields)
+                util.ping_token(self.waiter_url, token_name)
+            util.post_token(self.waiter_url, token_name, token_fields)
+            service_id = util.ping_token(self.waiter_url, token_name, 200)
+            goal_fn = lambda insts: min_instances == len(insts['active-instances']) and \
+                                    0 == len(insts['failed-instances']) and \
+                                    0 == len(insts['killed-instances'])
+            util.wait_until_routers_service(self.waiter_url, service_id, lambda service: goal_fn(service['instances']))
+            instances = util.instances_for_service(self.waiter_url, service_id)
+            logging.info(f'instances: {instances}')
+            possible_instances = get_possible_instances_fn(service_id, instances)
+            signal_flags = [signal_flags] if signal_flags else []
+            if test_instance:
+                possible_instances = possible_instances[0:1]
+                signal_dest = possible_instances[0]['id']
+                signal_flags.append('-i')
+            elif test_service:
+                signal_dest = service_id
+                signal_flags.append('-s')
+            else:
+                signal_dest = token_name
+            if signal_type == 'soft-delete':
+                signal_type_output = 'sigterm'
+            elif signal_type == 'hard-delete':
+                signal_type_output = 'sigkill'
+            else:
+                signal_type_output = ''
+            cp = cli.signal(self.waiter_url, signal_type, signal_dest, signal_flags=' '.join(signal_flags), stdin=stdin)
+            stdout = cli.stdout(cp)
+            self.assertEqual(0, cp.returncode, cp.stderr)
+            self.assertIn(f'Successfully sent {signal_type_output} to', stdout)
+            kill_fn = lambda insts: (min_instances - 1) == len(insts['active-instances']) and \
+                                    1 == len(insts['killed-instances'])
+            util.wait_until_routers_service(self.waiter_url, service_id, lambda service: kill_fn(service['instances']))
+            killed_instances = util.specific_instances_for_service(self.waiter_url, service_id, 'killed-instances')
+            killed_instance_id = killed_instances[0]['id']
+            active_instances = util.specific_instances_for_service(self.waiter_url, service_id,'active-instances' )
+            self.assertNotIn(killed_instance_id, active_instances, "instance is in active instances")
```

Comment:
`killed_instance_id` is a string id, where as `active_instances` is a list of maps. The type for the second argument needs to be list of strings (the ids).

Reply:
This assertion feels wrong. We need to check the instance ID we passed is present in killed instances and not present in active instances.